chartfx-chart/src/test/java/de/gsi/chart/ui/css/CssPropertyFactoryTest.java

```diff
@@ -0,0 +1,161 @@
+package de.gsi.chart.ui.css;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.testfx.api.FxAssert.verifyThat;
+
+import java.util.List;
+
+import javafx.css.CssMetaData;
+import javafx.css.PseudoClass;
+import javafx.css.StyleConverter;
+import javafx.css.Styleable;
+import javafx.css.StyleableBooleanProperty;
+import javafx.css.StyleableDoubleProperty;
+import javafx.css.StyleableFloatProperty;
+import javafx.css.StyleableIntegerProperty;
+import javafx.css.StyleableLongProperty;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableStringProperty;
+import javafx.scene.Scene;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testfx.api.FxRobot;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+
+import de.gsi.chart.ui.utils.JavaFXInterceptorUtils.SelectiveJavaFxInterceptor;
+
+/**
+ * Tests {@link de.gsi.chart.ui.css.CssPropertyFactory }
+ *
+ * @author akrimm
+ */
+@ExtendWith(ApplicationExtension.class)
+@ExtendWith(SelectiveJavaFxInterceptor.class)
+public class CssPropertyFactoryTest {
+    private final TestStyleable styleable = new TestStyleable();
+
+    @Start
+    public void start(Stage stage) {
+        // we need to add the styleable to a scene and display it or the css will not be evsaluated
+        final VBox root = new VBox(styleable);
+        final Scene scene = new Scene(root, 100, 120);
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    @Test
+    public void testApplyingStyle(FxRobot fxRobot) {
+        verifyThat(styleable.getMinHeight(), equalTo(-1.0)); // parent property
+        verifyThat(styleable.getTestDouble(), equalTo(4.2));
+        verifyThat(styleable.getTestInt(), equalTo(42));
+        verifyThat(styleable.getTestString(), equalTo("test"));
+        verifyThat(styleable.getTestBool(), equalTo(false));
+        verifyThat(styleable.getTestObject(), equalTo(Color.PINK));
+        verifyThat(styleable.getTestEnum(), equalTo(TestEnum.ENABLED));
+        verifyThat(styleable.getTestEnumP(), equalTo(TestEnum.DISABLED));
+        verifyThat(styleable.getPseudoClassStates(), Matchers.contains(PseudoClass.getPseudoClass("disabled")));
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Matchers.contains' due to existing static import 'org.hamcrest.Matchers.contains'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5741094)